perfkitbenchmarker/benchmarks/fio_benchmark.py

```diff
@@ -61,53 +264,68 @@ def GetInfo():
+     at the same path on the local machine.
 
   Args:
     benchmark_spec: The benchmark specification. Contains all data that is
         required to run the benchmark.
+
   """
-  vms = benchmark_spec.vms
-  vm = vms[0]
+
+  if FLAGS.fio_jobfile:
+    ignored_flags = {'--' + flag_name
+                     for flag_name in FLAGS_IGNORED_FOR_CUSTOM_JOBFILE
+                     if FLAGS[flag_name].present}
+
+    if ignored_flags:
+      logging.warning('Fio job file specified. Ignoring options "%s"',
+                      ', '.join(ignored_flags))
+
+  vm = benchmark_spec.vms[0]
   logging.info('FIO prepare on %s', vm)
   vm.Install('fio')
-  file_path = data.ResourcePath(flags.FLAGS.fio_jobfile)
-
-  disk_size_kb = vm.GetDeviceSizeFromPath(vm.GetScratchDir())
-  amount_memory_kb = vm.total_memory_kb
-  if disk_size_kb < amount_memory_kb * flags.FLAGS.memory_multiple:
-    logging.error('%s must be larger than %dx memory"',
-                  vm.GetScratchDir(),
-                  flags.FLAGS.memory_multiple)
-    # TODO(user): exiting here is probably not the correct behavor.
-    #    When FIO is run across a data set which is too not considerably
-    #    larger than the amount of memory then the benchmark results will be
-    #    invalid. Once the benchmark results are returned from Run() an
-    #    invalid (or is that rather a 'valid' flag should be added.
-    exit(1)
+
+  # Fill the disk or file we're using
+  disk = vm.scratch_disks[0]
+  device_path = disk.GetDevicePath()
+  mount_point = disk.mount_point
   if FLAGS.against_device:
-    device_path = vm.scratch_disks[0].GetDevicePath()
-    logging.info('Umount scratch disk on %s at %s', vm, device_path)
-    vm.RemoteCommand('sudo umount %s' % vm.GetScratchDir())
-    logging.info('Fill scratch disk on %s at %s', vm, device_path)
-    command = (
-        ('sudo %s --filename=%s --ioengine=libaio '
-         '--name=fill-device --blocksize=512k --iodepth=64 '
-         '--rw=write --direct=1 --size=%s') %
-        (fio.FIO_PATH, device_path, FLAGS.device_fill_size))
+    logging.info('Umount scratch disk on %s at %s', vm, mount_point)
+    vm.RemoteCommand('sudo umount %s' % mount_point)
+
+  if FLAGS.device_fill_size is not '0':
+    if FLAGS.against_device:
+      fill_path = device_path
+      fill_size = FLAGS.device_fill_size
+    else:
+      fill_path = posixpath.join(mount_point, DEFAULT_TEMP_FILE_NAME)
+      if FLAGS['disk_fill_size'].present:
+        fill_size = FLAGS.disk_fill_size
+      else:
+        # Default to 90% of capacity because the file system will add
+        # some overhead.
+        fill_size = str(DISK_USABLE_SPACE_FRACTION *
+                        1000 * disk.disk_size) + 'M'
+
+    logging.info('Fill file %s on %s', fill_path, vm)
+    command = GenerateFillCommand(fio.FIO_PATH,
+                                  fill_path,
+                                  fill_size)
     vm.RemoteCommand(command)
```

Comment:
The default behavior is changed to if we do nothing with the device_fill_size, we would fill the device? Please at least document the default behavior to avoid confusions.